core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -740,6 +743,7 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
+    successfulTaskDurations.remove(taskInfos(tid).duration)
```

Comment:
You mean for shuffle map tasks, where the task completed successfully but then the executor was lost later, so we re-run the task?  That case seems OK to leave the task time in, because it's still a valid task completion time.